### ISSUE TITLE
Feature roundup: Looking like a real blockchain

### ISSUE THREAD

**Issue by tcoulter** [ConsenSys-archive/ganache-cli-archive#127]

**Comment by tcoulter** [comment]
This is a feature roundup. It's meant to provide a discussion point for various, similar features that make the TestRPC's block processing look and feel like the real thing. This includes, but is not limited to: #66 and #31

The goal is to provide more realistic block times. Right now, whenever you make a transaction the TestRPC "insta-mines" it -- meaning, by the time the request finishes processing and you get a transaction id back, the transaction itself has already been processed. Being able to specify the block time, for instance, makes the TestRPC seem more like a normal blockchain.

The issue with this feature is it requires a big rewrite of block processing. Right now every transaction is set up to insta-mine. In fact, blocks are created along with transactions. In order to act like a more realistic blockchain, we would need to disjoin transaction processing and block processing, which is a lot of work.

This is a good feature to have, and I wanted to consolidate it here to accurately communicate the amount of work required.

**Comment by tcoulter** [comment]
Update: There has been significant refactoring to allow different strategies. Now those strategies need to be implemented. Looking for volunteers if anyone is interested.

**Comment by tcoulter** [comment]
With `miner_start` and `miner_stop` implemented on `develop`, the last thing that needs to happen until we can look like a "real blockchain" is to separate the interval mining and instamine components. The current behavior if you use `--blocktime` is to instamine on transactions but _also mine_ on an interval. It's now possible to remove the instamine component when `--blocktime` is used such that transactions are queued until the block is mined on the interval.

**Comment by tcoulter** [comment]
- [x] Removing instamining component when interval mining.
- [x] Write tests for interval mining such that we ensure blocks are processed.
- [x] Print transaction errors to logs since we can't handle them ourselves (technically already done)

**Comment by tcoulter** [comment]
- [x] Write a test showing error is printed to log since it's conditional and not easily verified
- [x] `miner_stop` should pause interval mining; `miner_start` should start it up again.

**Comment by tcoulter** [comment]
All done. On `master`, what what!? 💃

**Comment by tcoulter** [comment]
FYI: This commit: https://github.com/ethereumjs/testrpc/commit/efa8d545c0fe7de3bacfaeeef0240af11fbba128
